src/hooks/NoInternetToastProvider.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { createContext, useEffect, useRef, useState } from 'react';
+import { useNetInfo } from '@react-native-community/netinfo';
+import { t } from 'i18next';
+import Toast, { ToastProps } from 'react-native-toast-message';
+
+export const NoInternetToastContext = createContext({});
+
+export const NoInternetToastProvider = ({
+  children,
+  toastOverrides,
+  forceShowToast,
+}: {
+  children?: React.ReactNode;
+  toastOverrides?: ToastProps;
+  forceShowToast?: boolean;
+}) => {
+  const { isConnected } = useNetInfo();
+  const hiddenByAPI = useRef<boolean>(false);
+  const [forceRefresh, setForceRefresh] = useState<boolean>(false);
+
+  useEffect(() => {
+    if (isConnected === false || forceShowToast === false) {
```

Comment:
Should this be `&&` instead of `||` ?  That way, `forceShowToast` will always force execution into this block.

Reply:
`isConnected` will always be true so if we make this an `&&` then we will never enter this block.

Reply:
I rearrange the `_isConnected` to be the first evaluated to be sure the overwritten value is always honored first.